iosApp/iosApp/Pages/StopDetails/StopDetailsPage.swift

```diff
@@ -118,19 +92,15 @@ struct StopDetailsPage: View {
                     try? await Task.sleep(for: .seconds(5))
                 }
             }
-            .onDisappear { leavePredictions() }
+            .onDisappear { stopDetailsVM.leavePredictions() }
             .withScenePhaseHandlers(
                 onActive: {
-                    if let predictionsByStop,
-                       predictionsRepository
-                       .shouldForgetPredictions(predictionCount: predictionsByStop.predictionQuantity()) {
-                        self.predictionsByStop = nil
-                    }
-                    joinPredictions(stopId)
+                    stopDetailsVM.returnFromBackground()
+                    joinPredictions()
                 },
-                onInactive: leavePredictions,
-                onBackground: {
-                    leavePredictions()
+                onInactive: stopDetailsVM.leavePredictions,
+                onBackground: { @MainActor in
```

Comment:
I'm curious if it'd work to make these callbacks `@MainActor` in `withScenePhaseHandlers` instead of specifying that this specific one is `@MainActor` - I'd guess that several of these need to be on the main thread and none of them need to be off the main thread - but if that doesn't work then this seems like it's fine.

Reply:
Yeah good call, updated.